external/fv3fit/fv3fit/reservoir/autoencoder.py

```diff
@@ -0,0 +1,43 @@
+import os
+import tensorflow as tf
+from fv3fit._shared import get_dir, put_dir
+
+
+class Autoencoder(tf.keras.Model):
+    _ENCODER_NAME = "encoder.tf"
+    _DECODER_NAME = "decoder.tf"
+
+    def __init__(self, encoder, decoder):
+        super(Autoencoder, self).__init__()
+        self.n_latent_dims = encoder.layers[-1].output.shape[-1]
+        self.encoder = encoder
+        self.decoder = decoder
+
+    def call(self, x):
+        encoded = self.encoder(x)
```

Comment:
Is there a reason you call `self.encoder.__call__` in `call` but `self.encoder.predict` in `encode`?

Reply:
There's some differences between `Model.call` and `Model.predict` (https://keras.io/getting_started/faq/#whats-the-difference-between-model-methods-predict-and-call) so to be safe I just tried to be consistent. I figured if calling encode/decode the user is always just wanting the output values, versus the other potential uses of `__call__`.

Reply:
Optional suggestion: rename this method to `__call__` and add a `.predict` method which uses `.predict` on the underlying encoder and decoder.

Reply:
As per https://stackoverflow.com/a/59407927/5597448 I opted to keep the current version (define `call`, not `__call__`)